src/octopeer-github/main/Database/EventObject/EventFactory.ts

```diff
@@ -49,35 +48,39 @@ abstract class EventFactory {
      * @param position_y    the y-position of the mouse relative to the document.
      * @param viewport_x    the x-position of the mouse relative to the viewport.
      * @param viewport_y    the y-position of the mouse relative to the viewport.
-     * @param timestamp     the timestamp of the event.
+     * @param created_at    the timestamp at which the event was created. Defaults to the current timestamp.
      * @returns {MousePositionEvent} A MousePositionEvent object that can be posted to the database.
      */
     public static mousePosition(position_x: number, position_y: number,
-                                viewport_x: number, viewport_y: number, timestamp: UnixTimestamp): MousePositionEvent {
-        return {position_x: position_x, position_y: position_y, timestamp: timestamp,
+                                viewport_x: number, viewport_y: number, created_at = EventFactory.getTime()): MousePositionEvent {
+        return {created_at: created_at, position_x: position_x, position_y: position_y,
                 viewport_x: viewport_x, viewport_y: viewport_y};
     }
 
     /**
      * Creates a MouseScrollEvent object.
      * @param viewport_x    the x-position of the mouse relative to the viewport.
      * @param viewport_y    the y-position of the mouse relative to the viewport.
-     * @param timestamp     the timestamp of the event.
+     * @param created_at    the timestamp at which the event was created. Defaults to the current timestamp.
      * @returns {MouseScrollEvent} A MouseScrollEvent object that can be posted to the database.
      */
-    public static mouseScroll(viewport_x: number, viewport_y: number, timestamp: UnixTimestamp): MouseScrollEvent {
-        return {timestamp: timestamp, viewport_x: viewport_x, viewport_y: viewport_y};
+    public static mouseScroll(viewport_x: number, viewport_y: number, created_at = EventFactory.getTime()): MouseScrollEvent {
+        return {created_at: created_at, viewport_x: viewport_x, viewport_y: viewport_y};
     }
 
     /**
      * Creates a WindowResolutionEvent object.
      * @param width         the new width of the window.
      * @param height        the new height of the window.
-     * @param timestamp     the timestamp of the event.
+     * @param created_at    the timestamp at which the event was created. Defaults to the current timestamp.
      * @returns {WindowResolutionEvent} A WindowResolutionEvent object that can be posted to the database.
      */
-    public static windowResolution(width: number, height: number, timestamp: UnixTimestamp): WindowResolutionEvent {
-        return {height: height, timestamp: timestamp, width: width};
+    public static windowResolution(width: number, height: number, created_at = EventFactory.getTime()): WindowResolutionEvent {
+        return {created_at: created_at, height: height, width: width};
+    }
+
+    public static getTime() {
```

Comment:
I should've stated that just a little clearer in my PR summary (will edit), but Aaron's database runs on timestamps in seconds. In order to be consistent, I rewrote [the documentation of `Unixtimestamp`](https://github.com/thervh70/ContextProject_RDD/pull/112/files#diff-d966d67a48ca3c009062e7952dfed721R21) to match this. Because `new Date().getTime()` returns a timestamp in milliseconds, you have to `/ 1000` to get to seconds (duh), but by calling `EventFactory.getTime()` you do not have to worry about seconds or milliseconds anymore. 
Will update the documentation :)